util/dvsim/SlurmLauncher.py

```diff
@@ -0,0 +1,137 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+import os
+import shlex
+import subprocess
+
+from Launcher import ErrorMessage, Launcher, LauncherError
+
+
+SLURM_QUEUE = os.environ.get("SLURM_QUEUE", "hw-m")
+SLURM_MEM = os.environ.get("SLURM_MEM", "16G")
+SLURM_MINCPUS = os.environ.get("SLURM_MINCPUS", "8")
+SLURM_TIMEOUT = os.environ.get("SLURM_TIMEOUT", "240")
+SLURM_CPUS_PER_TASK = os.environ.get("SLURM_CPUS_PER_TASK", "8")
+SLURM_SETUP_CMD = os.environ.get("SLURM_SLURM_SETUP_CMD", "")
+
+
+class SlurmLauncher(Launcher):
+    # Misc common SlurmLauncher settings.
+    max_odirs = 5
+
+    def __init__(self, deploy):
+        '''Initialize common class members.'''
+
+        super().__init__(deploy)
+
+        # Popen object when launching the job.
+        self.process = None
+        self.slurm_log_file = self.deploy.get_log_path() + '.slurm'
+
+    def _do_launch(self):
+        # replace the values in the shell's env vars if the keys match.
+        exports = os.environ.copy()
+        exports.update(self.deploy.exports)
+
+        # Clear the magic MAKEFLAGS variable from exports if necessary. This
+        # variable is used by recursive Make calls to pass variables from one
+        # level to the next. Here, self.cmd is a call to Make but it's
+        # logically a top-level invocation: we don't want to pollute the flow's
+        # Makefile with Make variables from any wrapper that called dvsim.
+        if 'MAKEFLAGS' in exports:
+            del exports['MAKEFLAGS']
+
+        self._dump_env_vars(exports)
+
+        try:
+            with open(self.slurm_log_file, 'w') as out_file:
+                out_file.write("[Executing]:\n{}\n\n".format(self.deploy.cmd))
+                out_file.flush()
+
+                # Add a command delimiter if necessary
```

Comment:
I'd create the slurm_cmd outside the try block so the try just does IO and creates the process.